### DISCUSSION TITLE
'Session ID unknown' after deploying  to test environment with node.

### DISCUSSION THREAD

**Discussion by Xiaotou-Sophia** [socketio/socket.io#4400] [Q&A] [answered]
Description：
It seems works fine in local development environment, and I got the right response. So I deploy all my client-end  and server -end to test environment, but I got a 400 response with 'Session ID unknown'.   BTW, there  is only one node instance in the test environment, and I haven't used redis or other database in my socket related business yet.

Dev Environment: 
- Client: 
  -  node : v14.0.0
  - "socket.io-client": "^4.5.1"
  - usage:   `socket.current = io(`${window.location.origin}`); `

- Server:
  - node: v14.0.0
  - "socket.io": "^4.5.1"
  -  "express": "^4.17.1"
  - usage: 
  ```
  const socket = require('socket.io');
  const io = socket(server);
  ````

Nginx config:
```
proxy_http_version 1.1;
proxy_set_header Upgrade $http_upgrade;
proxy_set_header Connection "Upgrade";

proxy_set_header Host $http_host;
proxy_set_header X-Client-Proto $scheme;
proxy_set_header X-Client-Proto-Ver $server_protocol;
proxy_set_header X-Real-IP $remote_addr;
proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
```

- Tried solution but  did not solve my question: 
After checked many issues and responses , I tried add transports config like `transports:[ 'websocket', 'polling'],`  in both my client-end and server-end , all of sudden, it doesn't  work in my Dev environment, and it only work with `transports:[ 'polling'],`.  But still , didn't work for my test environment.

**Comment by Xiaotou-Sophia** [comment]
issue like this:
`{"code":1,"message":"Session ID unknown"}`

<img width="939" alt="image" src="https://user-images.githubusercontent.com/18388408/174611699-9ac99a27-abcb-423b-b444-6df6c4c2611b.png">

**Comment by darrachequesne** [comment, marked as answer]
Hi! That's weird, I think the only way to get a `session ID is unknown` error is when you use several server instances: https://github.com/socketio/engine.io/blob/917d1d29e13f2e8f523c3738f6413f67b587aebe/lib/server.ts#L225-L229

Are you able to reliably reproduce?

Also, could you please try with `transports: ['websocket']` on the client side?

**Reply by Xiaotou-Sophia** [reply]
Hi Damien,  I'm glad to hear from you  and now it works in my test environment. But it so tricky that after I added `transports: ['websocket']` , my local dev environment got nothing  from network and of course no socket response. So I added a logical judgment to distinguish the environment in the client side. But still, I wonder what's going on ? Why is such a difference.
